### PR TITLE
Upgrade blob inventory policy API to 2021-04-01

### PR DESCRIPTION
**Pull request by magodo**
This PR upgrades the blob inventory policy API from 2021-01-01 to 2021-04-01. As is stated in https://github.com/Azure/azure-rest-api-specs/issues/16696, there are several breaking changes makes the old instances not able to be managed by the new API version. Therefore, this PR deprecates the top level `storage_container_name` and introduced some new properties.

## Test

```bash
💤 TF_ACC=1 go test -v -timeout=20h ./internal/services/storage -run=TestAccStorageBlobInventoryPolicy_update
=== RUN   TestAccStorageBlobInventoryPolicy_update
=== PAUSE TestAccStorageBlobInventoryPolicy_update
=== CONT  TestAccStorageBlobInventoryPolicy_update
--- PASS: TestAccStorageBlobInventoryPolicy_update (537.62s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/storage       537.661s
```

### CODE REVIEW

**Review by katbyte — APPROVED** [review]
Thanks @magodo - LGTM 🏗️

**Review by tombuildsstuff — COMMENTED** [review]
A couple of comments about the migration but this otherwise LGTM 👍

**Comment by katbyte** [comment]
@magodo - this is good to merge once the merge conflict is resolved

**Pull request merged**
